context/app/static/js/components/Detail/provenance/ProvGraph/ProvGraph.jsx

```diff
@@ -1,11 +1,30 @@
-import React from 'react';
+import React, { useContext } from 'react';
 import PropTypes from 'prop-types';
+import Button from '@material-ui/core/Button';
 
 import SectionItem from 'js/components/Detail/SectionItem';
 import { LightBlueLink } from 'js/shared-styles/Links';
+import { AppContext } from 'js/components/Providers';
+import { getImmediateDescendantProv } from 'js/hooks/useImmediateDescendantProv';
+import useProvenanceStore from 'js/stores/useProvenanceStore';
 import ProvVis from '../ProvVis';
 import { StyledTypography, FlexPaper } from './style';
 import '@hms-dbmi-bgm/react-workflow-viz/dist/react-workflow-viz.min.css';
+import ProvData from '../ProvVis/ProvData';
+
+function createStepNameSet(steps) {
+  const nameSet = new Set([]);
+  steps.forEach((step) => nameSet.add(step.name));
+  return nameSet;
+}
+
+function removeExistingSteps(steps, newSteps) {
+  const nameSet = createStepNameSet(steps);
+  const uniqueNewSteps = [...newSteps].filter((step) => !nameSet.has(step.name));
```

Comment:
```suggestion
  const uniqueNewSteps = newSteps.filter((step) => !nameSet.has(step.name));
```
I might be missing something?